model/perf.go

```diff
@@ -637,3 +638,151 @@ func (r *PerformanceResults) FindOutdatedRollups(ctx context.Context, name strin
 
 	return errors.WithStack(it.Close(ctx))
 }
+
+func GetTimeSeriesIds(ctx context.Context, db *mongo.Database) (*mongo.Cursor, error) {
+	pipeline := bson.A{
+		bson.M{
+			"$match": bson.M{
+				"info.order": bson.M{
+					"$exists": true,
+				},
+				"info.mainline":  true,
+				"info.project":   bson.M{"$exists": true},
+				"info.variant":   bson.M{"$exists": true},
+				"info.task_name": bson.M{"$exists": true},
+				"info.test_name": bson.M{"$exists": true},
+			},
+		},
+		bson.M{
+			"$unwind": "$rollups.stats",
+		},
+		bson.M{
+			"$group": bson.M{
+				"_id": bson.M{
+					"task":        "$info.task_name",
+					"variant":     "$info.variant",
+					"project":     "$info.project",
+					"test":        "$info.test_name",
+					"measurement": "$rollups.stats.name",
+				},
+			},
+		},
+	}
+	return db.Collection(perfResultCollection).Aggregate(ctx, pipeline)
+}
+
+type TimeSeriesId struct {
+	Id struct {
+		Project     string `bson:"project"`
+		Variant     string `bson:"variant"`
+		Task        string `bson:"task"`
+		Test        string `bson:"test"`
+		Measurement string `bson:"measurement"`
+	} `bson:"_id"`
+}
+
+type TimeSeriesEntry struct {
+	PerfResultID string `bson:"id"`
+	Value        float64
+	Order        int
+}
+
+type TimeSeries struct {
+	TimeSeriesId TimeSeriesId      `bson:"_id"`
+	Data         []TimeSeriesEntry `bson:"times_series"`
+}
+
+func makeTimeSeriesPipeline(timeSeriesId TimeSeriesId) []bson.M {
+	var pipeline []bson.M
+
+	pipeline = append(pipeline, bson.M{
+		"$match": bson.M{
+			"info.project":   timeSeriesId.Id.Project,
+			"info.variant":   timeSeriesId.Id.Variant,
+			"info.task_name": timeSeriesId.Id.Task,
+			"info.test_name": timeSeriesId.Id.Test,
+		},
+	})
+
+	pipeline = append(pipeline, bson.M{
+		"$match": bson.M{
+			"info.order": bson.M{
+				"$exists": true,
+			},
+			"info.mainline": true,
+		},
+	})
+
+	pipeline = append(pipeline, bson.M{
+		"$unwind": "$rollups.stats",
+	})
+
+	pipeline = append(pipeline, bson.M{
+		"$match": bson.M{
+			"rollups.stats.name": timeSeriesId.Id.Measurement,
+		},
+	})
+
+	pipeline = append(pipeline, bson.M{
+		"$project": bson.M{
+			"measurement": "$rollups.stats.name",
+			"value":       "$rollups.stats.val",
+			"order":       "$info.order",
+		},
+	})
+
+	pipeline = append(pipeline, bson.M{
+		"$group": bson.M{
+			"_id": "$measurement",
+			"time_series": bson.M{
+				"$push": bson.M{
+					"id":    "$_id",
+					"value": "$value",
+					"order": "$order",
+				},
+			},
+		},
+	})
```

Comment:
why not just do: 

`return []bson.M{{<...>}, {<...>}}`?

The appends make it harder to read and don't serve a purpose.